### ISSUE TITLE
[IntelliJ][ReportedByUser] MS Azure Explorer - Error Loading Web Apps

### ISSUE THREAD

**Issue by LeftistTachyon** [microsoft/azure-tools-for-java#3900]
IntelliJ build version: 2019.3.1 IC-193.5662.53
OS: Windows 10
JDK: JetBrains s.r.o 11.0.5
Plugin version: 3.31.0-2019.3
Error message: 
```
java.lang.RuntimeException: com.microsoft.azuretools.adauth.AuthException: AD Auth token endpoint returned HTTP status code 400. Error info: {"error":"invalid_request","error_description":"AADSTS90002: Tenant 'null' not found. This may happen if there are no active subscriptions for the tenant. Check to make sure you have the correct tenant ID. Check with your subscription administrator.\r\nTrace ID: 107f3648-fcc7-4895-a75a-3728f7f0c000\r\nCorrelation ID: 809dd951-438d-4376-993f-c667f6d19257\r\nTimestamp: 2020-01-06 02:57:22Z","error_codes":[90002],"timestamp":"2020-01-06 02:57:22Z","trace_id":"107f3648-fcc7-4895-a75a-3728f7f0c000","correlation_id":"809dd951-438d-4376-993f-c667f6d19257","error_uri":"https://login.microsoftonline.com/error?code=90002"}
	at rx.exceptions.Exceptions.propagate(Exceptions.java:57)
	at rx.observables.BlockingObservable.blockForSingle(BlockingObservable.java:463)
	at rx.observables.BlockingObservable.single(BlockingObservable.java:340)
	at com.microsoft.azure.management.appservice.implementation.WebAppsInner.list(WebAppsInner.java:1847)
	at com.microsoft.azure.
```

Just before this error occurred, I attempted to deploy my Java EE Server Application (for Tomcat 9.x) to an Azure App Service. The console gave this error:
```
com.microsoft.azuretools.adauth.AuthException: AD Auth token endpoint returned HTTP status code 400. Error info: {"error":"invalid_request","error_description":"AADSTS90002: Tenant 'null' not found. This may happen if there are no active subscriptions for the tenant. Check to make sure you have the correct tenant ID. Check with your subscription administrator.\r\nTrace ID: 6d9bacf4-347a-4aaf-ae2d-682f1d68ab00\r\nCorrelation ID: 7b13e751-4b6b-48fe-aa7d-57c1db29f067\r\nTimestamp: 2020-01-06 02:56:59Z","error_codes":[90002],"timestamp":"2020-01-06 02:56:59Z","trace_id":"6d9bacf4-347a-4aaf-ae2d-682f1d68ab00","correlation_id":"7b13e751-4b6b-48fe-aa7d-57c1db29f067","error_uri":"https://login.microsoftonline.com/error?code=90002"}
```
After this error, I navigated to the Azure Explorer window on the side of IntelliJ (provided by the Azure Toolkit for IntelliJ) and double-clicked `Web Apps` in the menu under my subscription. Afterwards, the main error occured, whose stack trace is given above.

It appears that my token for Azure that was stored in IntelliJ had expired, and the plugin didn't know how to handle it.

**Comment by LeftistTachyon** [comment]
After #3901 and #3902 happened after following the same replication steps (hitting the refresh icon for the Azure Explorer), this exception was thrown again. This time, I caught the entire stack trace in case it's helpful.
```
java.lang.RuntimeException: com.microsoft.azuretools.adauth.AuthException: AD Auth token endpoint returned HTTP status code 400. Error info: {"error":"invalid_request","error_description":"AADSTS90002: Tenant 'null' not found. This may happen if there are no active subscriptions for the tenant. Check to make sure you have the correct tenant ID. Check with your subscription administrator.\r\nTrace ID: e3f1cdcd-6ccb-4b86-a420-d21bb2c8b700\r\nCorrelation ID: 6750f7d9-998a-4df0-b07c-4c02c6e108c8\r\nTimestamp: 2020-01-06 03:01:49Z","error_codes":[90002],"timestamp":"2020-01-06 03:01:49Z","trace_id":"e3f1cdcd-6ccb-4b86-a420-d21bb2c8b700","correlation_id":"6750f7d9-998a-4df0-b07c-4c02c6e108c8","error_uri":"https://login.microsoftonline.com/error?code=90002"}
	at rx.exceptions.Exceptions.propagate(Exceptions.java:57)
	at rx.observables.BlockingObservable.blockForSingle(BlockingObservable.java:463)
	at rx.observables.BlockingObservable.single(BlockingObservable.java:340)
	at com.microsoft.azure.management.appservice.implementation.WebAppsInner.list(WebAppsInner.java:1847)
	at com.microsoft.azure.management.resources.fluentcore.arm.collection.implementation.TopLevelModifiableResourcesImpl.list(TopLevelModifiableResourcesImpl.java:118)
	at com.microsoft.azuretools.core.mvp.model.webapp.AzureWebAppMvpModel.listWebApps(AzureWebAppMvpModel.java:569)
	at com.microsoft.azuretools.core.mvp.model.webapp.AzureWebAppMvpModel.lambda$null$1(AzureWebAppMvpModel.java:528)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OperatorSubscribeOn$SubscribeOnSubscriber.call(OperatorSubscribeOn.java:100)
	at rx.internal.schedulers.CachedThreadScheduler$EventLoopWorker$1.call(CachedThreadScheduler.java:230)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.microsoft.azuretools.adauth.AuthException: AD Auth token endpoint returned HTTP status code 400. Error info: {"error":"invalid_request","error_description":"AADSTS90002: Tenant 'null' not found. This may happen if there are no active subscriptions for the tenant. Check to make sure you have the correct tenant ID. Check with your subscription administrator.\r\nTrace ID: e3f1cdcd-6ccb-4b86-a420-d21bb2c8b700\r\nCorrelation ID: 6750f7d9-998a-4df0-b07c-4c02c6e108c8\r\nTimestamp: 2020-01-06 03:01:49Z","error_codes":[90002],"timestamp":"2020-01-06 03:01:49Z","trace_id":"e3f1cdcd-6ccb-4b86-a420-d21bb2c8b700","correlation_id":"6750f7d9-998a-4df0-b07c-4c02c6e108c8","error_uri":"https://login.microsoftonline.com/error?code=90002"}
	at com.microsoft.azuretools.adauth.AuthContext.sendRequest(AuthContext.java:260)
	at com.microsoft.azuretools.adauth.AuthContext.access$100(AuthContext.java:46)
	at com.microsoft.azuretools.adauth.AuthContext$1.getTokenWithRefreshToken(AuthContext.java:245)
	at com.microsoft.azuretools.adauth.CacheDriver.acquireTokenFromMrrt(CacheDriver.java:117)
	at com.microsoft.azuretools.adauth.CacheDriver.refreshEntryIfNecessary(CacheDriver.java:109)
	at com.microsoft.azuretools.adauth.CacheDriver.find(CacheDriver.java:55)
	at com.microsoft.azuretools.adauth.AuthContext.acquireTokenFromCache(AuthContext.java:213)
	at com.microsoft.azuretools.adauth.AuthContext.acquireToken(AuthContext.java:126)
	at com.microsoft.azuretools.authmanage.AdAuthManager.getAccessToken(AdAuthManager.java:73)
	at com.microsoft.azuretools.authmanage.RefreshableTokenCredentials.getToken(RefreshableTokenCredentials.java:65)
	at com.microsoft.azure.credentials.AzureTokenCredentials.getToken(AzureTokenCredentials.java:74)
	at com.microsoft.azure.credentials.AzureTokenCredentialsInterceptor.intercept(AzureTokenCredentialsInterceptor.java:36)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at com.microsoft.azure.management.resources.fluentcore.utils.ResourceManagerThrottlingInterceptor.intercept(ResourceManagerThrottlingInterceptor.java:54)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at com.microsoft.azure.management.resources.fluentcore.utils.ProviderRegistrationInterceptor.intercept(ProviderRegistrationInterceptor.java:43)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at com.microsoft.azuretools.telemetry.TelemetryInterceptor.intercept(TelemetryInterceptor.java:46)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at com.microsoft.rest.interceptors.BaseUrlHandler.intercept(BaseUrlHandler.java:43)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at com.microsoft.rest.interceptors.RequestIdHeaderInterceptor.intercept(RequestIdHeaderInterceptor.java:29)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:250)
	at okhttp3.RealCall.execute(RealCall.java:93)
	at retrofit2.OkHttpCall.execute(OkHttpCall.java:186)
	at retrofit2.adapter.rxjava.CallExecuteOnSubscribe.call(CallExecuteOnSubscribe.java:40)
	at retrofit2.adapter.rxjava.CallExecuteOnSubscribe.call(CallExecuteOnSubscribe.java:24)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
	at rx.Observable.subscribe(Observable.java:10423)
	at rx.Observable.subscribe(Observable.java:10390)
	at rx.observables.BlockingObservable.blockForSingle(BlockingObservable.java:443)
	... 15 more
Caused by: java.io.IOException: AD Auth token endpoint returned HTTP status code 400. Error info: {"error":"invalid_request","error_description":"AADSTS90002: Tenant 'null' not found. This may happen if there are no active subscriptions for the tenant. Check to make sure you have the correct tenant ID. Check with your subscription administrator.\r\nTrace ID: e3f1cdcd-6ccb-4b86-a420-d21bb2c8b700\r\nCorrelation ID: 6750f7d9-998a-4df0-b07c-4c02c6e108c8\r\nTimestamp: 2020-01-06 03:01:49Z","error_codes":[90002],"timestamp":"2020-01-06 03:01:49Z","trace_id":"e3f1cdcd-6ccb-4b86-a420-d21bb2c8b700","correlation_id":"6750f7d9-998a-4df0-b07c-4c02c6e108c8","error_uri":"https://login.microsoftonline.com/error?code=90002"}
	at com.microsoft.azuretools.adauth.HttpHelper.sendPostRequestAndDeserializeJsonResponse(HttpHelper.java:92)
	at com.microsoft.azuretools.adauth.AuthContext.sendRequest(AuthContext.java:256)
	... 58 more
```

**Comment by wezhang** [comment]
Thanks @LeftistTachyon ! @Flanker32 , could you help check it?

**Comment by Flanker32** [comment]
@LeftistTachyon Thanks for your report and really sorry for the late response, could you please share your auth method to auth the IntelliJ Toolkit? And could this issue be reproduced after signout and re-signin?

**Comment by LeftistTachyon** [comment]
I just use the default auth method (?), which is to first take a code, paste it into some Microsoft site, then the plugin takes care of the rest.

No, the issue can't be reproduced after signing out and back in. It only happens after a long period of inactivity/not using the plugin for a long time.

**Comment by Flanker32** [comment]
@LeftistTachyon Got it, thanks, seems our device login token expire and failed to refresh. I'll looking into it soon.

**Comment by Flanker32** [comment]
@LeftistTachyon Sorry to bother you again, so the IDE toolkit could works well when you auth Azure with toolkit manually, and the issue could happen only if the toolkit auth Azure with previous credential files?

**Comment by LeftistTachyon** [comment]
Right now my machine isn't in the glitched state, so please give me a couple of days for the token to expire. Sorry to keep you waiting.

**Comment by Flanker32** [comment]
@LeftistTachyon Never mind, and thanks a lot for your report and help so that we could build the toolkits better~

Besides, could you please help us check the where the tenant id is set correctly or just empty in configuration file `${user.home}\.AzureToolsForIntelliJ\subscriptionsDetails-at.json`? In case of configuration exception.

**Comment by LeftistTachyon** [comment]
The file is not empty and seems to have the correct data, except for the tokens. I'm not sure if it's correct or not, but it probably was at one point or another.

**Comment by vscjavaci** [comment]
(Experimental duplicate detection)
Thanks for submitting this issue. Please check if it is already covered by an existing one:
- error (Message from Azure: 'Index: 0' Size: 0' ) while login using Eclipse Azure Toolkit (#30)

**Comment by vscjavaci** [comment]
(Experimental duplicate detection)
Thanks for submitting this issue. Please check if it is already covered by an existing one:
-  MS Azure Explorer - Error Loading Web Apps (#3653)

**Comment by vscjavaci** [comment]
Close this issue due to duplicate error: Tenant 'null' not found. This may happen if there are no active subscriptions for the tenant.
